### DISCUSSION TITLE
How to prevent critical bugs

### DISCUSSION THREAD

**Discussion by mitosagi** [team-apm/apm#1017] [Ideas]
In my experience, most of the critical bugs can be identified by the following steps:

1. Uninstall apm.
2. Remove `%APPDATA%\AviUtl Package Manager`.
3. Install the `.exe` created by `yarn make`.
4. Test the following items
- [ ] Launch correctly.
- [ ] AviUtl data, package data, and script data are automatically loaded.
- [ ] Recommended (Batch) installation runs without problems.

I would like to note this procedure here as I would like to do this before releasing apm or updating the major version of electron.

**Comment by yumetodo** [comment]
In generally, similar to your pointed out, these 3 case should be tested.

1. clean install
2. manual update
3. auto update
